provider/oauth2/tests.py

```diff
@@ -597,6 +597,25 @@ def test_access_token_response_valid_token_type(self):
         token = self._login_authorize_get_token()
         self.assertEqual(token['token_type'], constants.TOKEN_TYPE, token)
 
+    def test_client_credentials_grant(self):
```

Comment:
Nit: 
1. Separating the tests here into individual test methods.
2. For comprehensiveness, add a test for invalid client_id with client_credentials grant_type.
3. So, in the end, you would have: `test_client_credentials_grant_success`, `test_client_credentials_grant_invalid_secret`, `test_client_credentials_grant_invalid_client_id`.